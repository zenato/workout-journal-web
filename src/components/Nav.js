import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import classNames from 'classnames/bind';
import styles from './Nav.scss';

const cx = classNames.bind(styles);

const Nav = ({ user, onLogout }) => {
  return (
    <nav className={cx('nav')}>
      <div className={cx('main')}>
        <ul>
          <li><NavLink exact to="/" activeClassName={cx('active')}>Home</NavLink></li>
          <li><NavLink to="/events" activeClassName={cx('active')}>Events</NavLink></li>
          <li><NavLink to="/posts" activeClassName={cx('active')}>Posts</NavLink></li>
        </ul>
      </div>
      {user && (
        <aside className={cx('tool')}>
            <a href="" onClick={onLogout}>Logout</a>
        </aside>
      )}
    </nav>
  );
};

Nav.propTypes = {
  user: PropTypes.shape({
    username: PropTypes.string,
  }),
  onLogout: PropTypes.func.isRequired,
};

export default Nav;
