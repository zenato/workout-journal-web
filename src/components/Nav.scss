@import 'common';

.nav {
  $color: #f2f2f2;

  display: flex;
  background-color: #323232;
  color: $color;
  padding: 10px;
  maring: 0;
  font-size: 1.2rem;

  .main {
    flex: 1;
  }

  .tool {
    display: flex;
    width: 200px;
    align-items: flex-end;
    justify-content: flex-end;

    a {
      font-size: 0.8em;
    }
  }


  a {
    color: $color;
    &:hover, &:active, &:visited {
      color: $color;
    }
  }

  li:not(:last-child) {
    float: left;
    margin-right: 10px;
  }

  .active {
    color: red !important;
    font-weight: bold;
  }
}
