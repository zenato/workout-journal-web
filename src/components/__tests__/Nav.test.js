import React from 'react';
import { shallow } from 'enzyme';
import Nav from '../Nav';

const props = {
  user: null,
  onLogout: () => {},
};

it('Render without `user` prop.', () => {
  const component = shallow(<Nav {...props} />);
  expect(component).toMatchSnapshot();
});

it('Render with `user` prop.', () => {
  const user = { username: 'username' };
  const component = shallow(<Nav {...props} user={user} />);
  expect(component).toMatchSnapshot();
});

it('Call `logout` prop when click logout link.', () => {
  const onLogout = jest.fn();
  const user = { username: 'username' };
  const component = shallow(<Nav {...props} user={user} onLogout={onLogout} />);
  component.find('aside a').simulate('click');
  expect(onLogout).toBeCalled();
});
