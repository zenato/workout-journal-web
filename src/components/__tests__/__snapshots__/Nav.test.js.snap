// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <nav
    className="nav"
>
    <div
        className="main"
    >
        <ul>
            <li>
                <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    exact={true}
                    to="/"
                >
                    Home
                </NavLink>
            </li>
            <li>
                <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/events"
                >
                    Events
                </NavLink>
            </li>
            <li>
                <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/posts"
                >
                    Posts
                </NavLink>
            </li>
        </ul>
    </div>
    <aside
        className="tool"
    />
</nav>,
  "nodes": Array [
    <nav
      className="nav"
>
      <div
            className="main"
      >
            <ul>
                  <li>
                        <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              exact={true}
                              to="/"
                        >
                              Home
                        </NavLink>
                  </li>
                  <li>
                        <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/events"
                        >
                              Events
                        </NavLink>
                  </li>
                  <li>
                        <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/posts"
                        >
                              Posts
                        </NavLink>
                  </li>
            </ul>
      </div>
      <aside
            className="tool"
      />
</nav>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Nav />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <nav
          className="nav"
>
          <div
                    className="main"
          >
                    <ul>
                              <li>
                                        <NavLink
                                                  activeClassName="active"
                                                  ariaCurrent="true"
                                                  exact={true}
                                                  to="/"
                                        >
                                                  Home
                                        </NavLink>
                              </li>
                              <li>
                                        <NavLink
                                                  activeClassName="active"
                                                  ariaCurrent="true"
                                                  to="/events"
                                        >
                                                  Events
                                        </NavLink>
                              </li>
                              <li>
                                        <NavLink
                                                  activeClassName="active"
                                                  ariaCurrent="true"
                                                  to="/posts"
                                        >
                                                  Posts
                                        </NavLink>
                              </li>
                    </ul>
          </div>
          <aside
                    className="tool"
          />
</nav>,
        "_debugID": 2,
        "_renderedOutput": <nav
          className="nav"
>
          <div
                    className="main"
          >
                    <ul>
                              <li>
                                        <NavLink
                                                  activeClassName="active"
                                                  ariaCurrent="true"
                                                  exact={true}
                                                  to="/"
                                        >
                                                  Home
                                        </NavLink>
                              </li>
                              <li>
                                        <NavLink
                                                  activeClassName="active"
                                                  ariaCurrent="true"
                                                  to="/events"
                                        >
                                                  Events
                                        </NavLink>
                              </li>
                              <li>
                                        <NavLink
                                                  activeClassName="active"
                                                  ariaCurrent="true"
                                                  to="/posts"
                                        >
                                                  Posts
                                        </NavLink>
                              </li>
                    </ul>
          </div>
          <aside
                    className="tool"
          />
</nav>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Nav />,
}
`;
